### DISCUSSION TITLE
Increasing Gas Limit - Truffle Side

### DISCUSSION THREAD

**Discussion by harish-kar** [ConsenSys-archive/truffle#5581] [General Q&A] [answered]
Is there a way to increase the gas limit on Truffle-Config to beyond 6721975? I know this is the default value but even if I specify a higher value on the config file for Development, I don't think it works. Now the reason why I think it fails is that I have a really complicated contract that reverts without any error message, but indicates that the cumulative gas used is 6529031 (which is almost the limit). The contract is correct because it works for smaller values but as I try to scale the gas consumption clearly goes up.

**Comment by apdan510** [comment]
Hey @harishk93 can you please share your Truffle config file with us? Thank you

**Reply by harish-kar** [reply]
```
module.exports = {
    networks: {
        development: {
            host: "127.0.0.1",
            port: 8545, // ganache
            gasPrice: 0,
            network_id: "*", // Match any network id
            websockets: true,
            baseFeePerGas: 0,
            gas:500000000000
            // maxPriorityFeePerGas: 20000000000000,
        }
    },
    plugins: ["truffle-contract-size"],
    compilers: {
        solc: {
            version: "0.7.0",
             settings: { // See the solidity docs for advice about optimization and evmVersion
                optimizer: {
                    enabled: true,
                    runs: 200
                }   
             }
        }
    }
};
```
This is my config file. I am running Ganache with the same limit.

**Reply by harish-kar** [reply]
`     Error: Transaction has been reverted by the EVM:
{
  "transactionHash": "0x1ca0e84ca96ef192bb3958122804aaac708b81ff92f011895001f97cc80e9e80",
  "transactionIndex": 0,
  "blockNumber": 97,
  "blockHash": "0x733e471b2c5e7a9c21103c380ec99ddd9f9270942941f2306d27158f71c4aa5d",
  "from": "0x2a31d2b10930b8f536fd52b6138ad40eab0d0b09",
  "to": "0x4459ca195e020c1f600dc77be8497f3994238d47",
  "cumulativeGasUsed": 6527921,
  "gasUsed": 6527921,
  "contractAddress": null,
  "logs": [],
  "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "status": false,
  "effectiveGasPrice": "0x0",
  "type": "0x0"
}
      at Object.TransactionError (/Users/harishk/.nvm/versions/node/v12.22.12/lib/node_modules/truffle/build/webpack:/node_modules/web3-core-helpers/src/errors.js:93:1)
      at Object.TransactionRevertedWithoutReasonError (/Users/harishk/.nvm/versions/node/v12.22.12/lib/node_modules/truffle/build/webpack:/node_modules/web3-core-helpers/src/errors.js:105:1)
      at /Users/harishk/.nvm/versions/node/v12.22.12/lib/node_modules/truffle/build/webpack:/node_modules/web3-eth/node_modules/web3-core-method/src/index.js:482:1
`

This is the error message that I receive on Truffle's terminal window. The error message on Ganache's terminal window is just Revert.

**Comment by harish-kar** [comment, marked as answer]
I fixed the error. It appears that I had updated the gas limit in every file, except for the Client javascript file where I was sending the transaction. I fixed it and it works perfectly.
